nixos/modules/system/boot/luksroot.nix

```
@@ -650,30 +639,25 @@ in
             '';
           };
 
-          keyFileSize = mkOption {
-            default = null;
-            example = 4096;
-            type = types.nullOr types.int;
-            description = lib.mdDoc ''
-              The size of the key file. Use this if only the beginning of the
-              key file should be used as a key (often the case if a raw device
-              or partition is used as key file). If not specified, the whole
-              `keyFile` will be used decryption, instead of just
-              the first `keyFileSize` bytes.
-            '';
-          };
-
-          keyFileOffset = mkOption {
-            default = null;
-            example = 4096;
-            type = types.nullOr types.int;
-            description = lib.mdDoc ''
-              The offset of the key file. Use this in combination with
-              `keyFileSize` to use part of a file as key file
-              (often the case if a raw device or partition is used as a key file).
-              If not specified, the key begins at the first byte of
-              `keyFile`.
+          keys = mkOption {
+            type = types.attrsOf (types.submodule (import ./luks/keys.nix {
+              inherit config lib;
+            }));
+            default = { };
+            example = literalExpression ''
+              {
+                default = {
+                  file = "/dev/sdb1";
```

Could unify the attrName as the file name, however, feel like this is a bit harder to read. However, having arbitrary attrNames does add some noise.
```suggestion
                "/dev/sdb1" = {
```